surfinBH/_fit_evaluators/fit_7dq4Emri.py

```diff
@@ -0,0 +1,267 @@
+
+        return q, chiA, chiB
+
+    #-------------------------------------------------------------------------
+    def _eval_wrapper(self, fit_key, q, chiA, chiB, **kwargs):
+        """ Evaluates the NRSur7dq4EmriRemnant model. """
+        chiA = np.array(chiA)
+        chiB = np.array(chiB)
+
+        # Warn/Exit if extrapolating
+        allow_extrap = kwargs.pop('allow_extrap', False)
+        if allow_extrap: warnings.warn("""Optional argument 'allow_extrap' is 
+        unused for this model. It works at arbitrarily large mass ratios.""")
+        self._check_param_limits(q, chiA, chiB, True)
+
+        x = np.concatenate(([q], chiA, chiB))
+
+        def eval_r_isco(chiz):
+            """
+            Estimate ISCO radius following Eq. (2.21) by Bardeen et al. 
+            (https://doi.org/10.1086/151796)
+            """
+ 
+            Z1 = 1+(1-chiz**2)**(1/3)*((1+chiz)**(1/3)+(1-chiz)**(1/3))
+            Z2 = (3*chiz**2+Z1**2)**(1/2)
+            r_ISCO = 3+Z2-np.sign(chiz)*((3-Z1)*(3+Z1+2*Z2))**(1/2)
+    
+            return r_ISCO
+
+        def eval_emriChif(q, chiA):
+            """
+            Estimate remnant spin for EMRI binary following Eq. (7) 
+            presented in THE PAPER.    
+            """
+            r_ISCO = eval_r_isco(chiA[2])
+            E_ISCO = np.sqrt(1-2/(3*r_ISCO))
+            L_ISCO = np.array([0,0,2/(3*3**(1/2))*(1+2*np.sqrt(3*r_ISCO-2))])
+                      
+            return chiA + 1/q*(L_ISCO-2*chiA*E_ISCO)  
+
+        def extrap_chif(x, fit_key):
+            """
+            The model evaluates remnant spin using GPR fit up to the upper bound
+            of the EMRI training region (q = 1000). At q > 1000, it returns the 
+            expected EMRI limit (Eq. (7) in THE PAPER). There is a transition 
+            function (Eq. (9) in THE PAPER) connecting smoothly these 
+            two regions. The transition takes place from q=1000 to q=2000.
+            """
+            x_i = x.copy()
+            x_i[0] = self.qEmriMax
+            
+            """This if statement defines the width of the transition region"""
+            if x[0] < 2*self.qEmriMax:
+                tmp = self._evaluate_fits(x_i, fit_key)
+                y_i, y_i_err = tmp.T[0], tmp.T[1]
+                y_f = eval_emriChif(2*self.qEmriMax, x[1:4])
+                y_f_err = 2*np.abs(x[1:4] - y_f)
+                y = (y_f - y_i) * np.sin(np.pi/(2*self.qEmriMax)* \
+                    (x[0] - self.qEmriMax))**2 + y_i
+                y_err = (y_f_err - y_i_err) * np.sin(np.pi/(2*self.qEmriMax)* \
+                    (x[0] - self.qEmriMax))**2 + y_i_err
+            else:
+                y = eval_emriChif(x[0], x[1:4])
+                y_err = 2*np.abs(x[1:4] - y)
+
+            return y, y_err
+
+        def eval_vector_fit(x, fit_key):
+            if fit_key == 'chif':
+                if x[0] > self.qEmriMax:
+                    fit_val, fit_err = extrap_chif(x, fit_key)
```

Comment:
extrap_chif seems like a bad name for this, given that you don't always extrap. Would eval_combined_chif work?

Reply:
If I move everything inside this function it could be also eval_chif